modules/460-log-shipper/hooks/internal/vector/transform/throttle.go

```diff
@@ -17,24 +17,84 @@ limitations under the License.
 package transform
 
 import (
+	"fmt"
+
 	"github.com/deckhouse/deckhouse/go_lib/set"
+	"github.com/deckhouse/deckhouse/modules/460-log-shipper/apis"
 	"github.com/deckhouse/deckhouse/modules/460-log-shipper/apis/v1alpha1"
+	"github.com/deckhouse/deckhouse/modules/460-log-shipper/hooks/internal/vrl"
+)
+
+const (
+	excludeField  = "exclude"
+	keyFieldField = "key_field"
 )
 
 // ThrottleTransform adds throttling to event's flow.
-func ThrottleTransform(rl v1alpha1.RateLimitSpec) *DynamicTransform {
+func ThrottleTransform(rl v1alpha1.RateLimitSpec) ([]apis.LogTransform, error) {
+	if rl.Excludes != nil && rl.KeyField != "" {
```

Comment:
Didn't get this. 

i think it is valid to have non empty excludes with an empty key field.